### PR TITLE
Add consistency check for deepcopy generated files

### LINKED ISSUE #1111: [Bug][CI] Add tests to ensure that the `zz_generated.deepcopy.go` is up-to-date.
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

ci

### What happened + What you expected to happen

When you update the CRD struct in `ray*_types.go` (e.g., `raycluster_types.go`), you should update the CRD YAML files and the generated code by executing `make build`. We have several [consistency checks in CI](https://github.com/ray-project/kuberay/blob/master/ray-operator/DEVELOPMENT.md#consistency-check), but we do not check `zz_generated.deepcopy.go`.

### Reproduction script

https://github.com/ray-project/kuberay/pull/1110/files#r1204973842

### Anything else

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by varungup90**
## Why are these changes needed?

Checks consistency for deepcopy generated files

## Related issue number
Closes #1111

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kevin85421** [comment]
cc @Yicheng-Lu-llll would you mind reviewing this PR?

File: .github/workflows/consistency-check.yaml [hunk 1]
```diff
@@ -30,6 +30,10 @@ jobs:
       - name: Verify Codegen
         working-directory: ./ray-operator
         run: ./hack/verify-codegen.sh
+
+      - name: Verify Generated Files
```
Comment (Yicheng-Lu-llll) [inline .github/workflows/consistency-check.yaml:34]: Just a nit-picky comment, we can have a better name here. 

-`Verify Generated Files`: showing it is used to verify the [aforementioned runtime.Object interface](https://book.kubebuilder.io/cronjob-tutorial/other-api-files.html) generated by kubebuilder. 

-`Verify Codegen`: showing it is used to verify things like clientset, lister, informer

**Comment by kevin85421** [comment]
There seems to be a conflict between "Verify Generated Files" and "Go-build-and-test / Lint (pull_request)". The linter expects `v1 "k8s.io/api/core/v1"` in zz_generated.deepcopy.go, but the "Verify Generated Files" step expects `"k8s.io/api/core/v1"`. Perhaps we need to fix the linter to align with the expected format.

**Comment by varungup90** [comment]
> There seems to be a conflict between "Verify Generated Files" and "Go-build-and-test / Lint (pull_request)". The linter expects `v1 "k8s.io/api/core/v1"` in zz_generated.deepcopy.go, but the "Verify Generated Files" step expects `"k8s.io/api/core/v1"`. Perhaps we need to fix the linter to align with the expected format.

Updated test-job.yaml file where goimports ignores zz_generated.deepcopy.go file. Verified it locally using github action `act -W .github/workflows/test-job.yaml`.

File: .github/workflows/test-job.yaml [hunk 2]
```diff
@@ -81,13 +81,13 @@ jobs:
         skip-pkg-cache: true
 
     - name: Run goimports
-      run: test -z "$(set -o pipefail && $(go env GOPATH)/bin/goimports -l apiserver/ ray-operator/ cli/ | tee goimports.out)" || { cat goimports.out && exit 1; }
+      run: test -z "$(set -o pipefail && $(go env GOPATH)/bin/goimports -l apiserver/ cli/ $(find ./ray-operator -name "*.go" | grep -v zz_generated.deepcopy.go) | tee goimports.out)" || { cat goimports.out && exit 1; }
 
     - name: Open this to see how to fix goimports if it fails
       run: |
-        echo "Run command 'goimports -w apiserver/ ray-operator/ cli/' to correct your code format."
+        echo "Run command 'goimports -w apiserver/ cli/ $(find ./ray-operator -name "*.go" | grep -v zz_generated.deepcopy.go)' to correct your code format."
```
Comment (kevin85421) [inline .github/workflows/test-job.yaml:88]: Could you add a comment explaining why you skip `zz_generated.deepcopy.go` and add this link https://github.com/kubernetes-sigs/controller-tools/issues/559?

File: ray-operator/hack/verify-generated-files.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+cleanup
+
+mkdir -p "${TMP_DIFFROOT}"
+cp -a "${DIFFROOT}"/* "${TMP_DIFFROOT}"
+
+make generate
+echo "diffing ${DIFFROOT} against freshly generated deepcopy files"
+ret=0
+diff -Naupr "${DIFFROOT}" "${TMP_DIFFROOT}" || ret=$?
+# cp -a "${TMP_DIFFROOT}"/* "${DIFFROOT}"
```
Comment (kevin85421) [inline ray-operator/hack/verify-generated-files.sh:27]: forget to remove?

File: .github/workflows/consistency-check.yaml (hunk 1 above)
Comment (varungup90) [reply .github/workflows/consistency-check.yaml:34]: I got the idea from Kubernetes repo (https://github.com/kubernetes/kubernetes/blob/v1.22.1/hack/verify-generated-files.sh). Please suggest better alternatives.

**Review by kevin85421 — APPROVED** [review]
LGTM.

To summarize this PR,

* Because the generated file `zz_generated.deepcopy.go` does not adhere to the `goimport` rule (as mentioned in this issue: https://github.com/kubernetes-sigs/controller-tools/issues/559), we have made the decision to exclude the `goimport` check for the `zz_generated.deepcopy.go` file. 

* Excluding the `goimport` check for the `zz_generated.deepcopy.go` file is fine because this PR will check the consistency for the generated files.

**Pull request merged**